packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -890,6 +894,17 @@ contract VotingV2 is
                 break;
             }
 
+            // If the request we're processing now is not the same round as the last index we processed successfully (not
+            // rolled), then we need to apply slashing because there's been a round change.
+            if (
```

Comment:
Note: this if was changed and moved from below the slash to above. We check that the previous index was the same round as the one we're about to process.

Note: `nextIndexToProcess - 1` is always the last processed index (i.e. the last index that was not rolled, but fully processed by this user). If nextIndexToProcess == 0, then there was no last processed index and therefore should never be any slashing to apply.

Reply:
ye, this makes a lot of sense, when combined with how you handle the `nextIndexToProcess` logic.